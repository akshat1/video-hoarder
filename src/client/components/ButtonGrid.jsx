import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import Style from './ButtonGrid.less';

const getButtonClassName = option => classNames({
  [Style.selected]: option.selected,
});

const getWrapperStyle = options => ({
  gridTemplateColumns: `repeat(${options.length}, 1fr)`,
});

const ButtonGrid = ({ options, onChange }) => (
  <div className={Style.wrapper} style={getWrapperStyle(options)}>
    <For each="option" of={options}>
      <button
        className={getButtonClassName(option)}
        disabled={option.disabled}
        onClick={() => onChange(option.value)}
        key={option.value}
      >
        {option.label}
      </button>
    </For>
  </div>
);

const ShapeOfOption = {
  className: PropTypes.string,
  disabled: PropTypes.boolean,
  label: PropTypes.string,
  selected: PropTypes.selected,
  value: PropTypes.any,
};

ButtonGrid.propTypes = {
  onChange: PropTypes.func,
  options: PropTypes.arrayOf(PropTypes.shape(ShapeOfOption)),
};

export default ButtonGrid;
