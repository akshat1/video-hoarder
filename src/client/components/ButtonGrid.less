@import (reference) "../style/vars.less";

.wrapper {
  display: grid;
  button {
    .border;
    // position: relative;
    // &:after {
    //   position: absolute;
    //   height: 10px;
    //   width: 10px;
    //   content: '';
    //   background-color: orange;
    //   left: 5px;
    //   top: 5px;
    //   border-radius: 5px;
    // }
    &:focus {
      box-shadow: @shadow-focus--inset;
    }
    &:hover {
      box-shadow: @shadow-hover--inset;
    }
    + button {
      border-left-color: darken(@color-border, 10%);
    }
  }
}

.selected {
  box-shadow: @shadow-active--inset;
}
