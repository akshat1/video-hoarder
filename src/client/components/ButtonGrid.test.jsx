import ButtonGrid from "./ButtonGrid.jsx";
import { shallow } from "enzyme";
import React from "react";

describe("components/ButtonGrid", () => {
  const options = [{
    className: "clFoo",
    disabled: true,
    label: "Foo",
    selected: true,
    value: "vFoo",
  }, {
    className: "clBar",
    label: "Bar",
    selected: true,
    value: "vBar",
  }, {
    className: "clBaz",
    label: "Baz",
    value: "vBaz",
  }, {
    className: "clQux",
    disabled: true,
    label: "Qux",
    value: "vQux",
  }];

  test("Renders and matches snapshot", () => {
    expect(shallow(<ButtonGrid options={options} onChange={() => 0} />)).toMatchSnapshot();
  });

  test("Calls onChange", () => {
    const onChange = jest.fn();
    const wrapper = shallow(<ButtonGrid options={options} onChange={onChange} />);
    const btnFoo = wrapper.findWhere(node => node.name() === "button" && node.text() === "Foo");
    const btnBaz = wrapper.findWhere(node => node.name() === "button" && node.text() === "Baz");
    btnFoo.simulate("click");
    expect(onChange).toHaveBeenCalledWith("vFoo");
    btnBaz.simulate("click");
    expect(onChange).toHaveBeenCalledWith("vBaz");
  });
});
