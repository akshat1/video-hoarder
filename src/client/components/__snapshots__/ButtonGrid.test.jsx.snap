// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/ButtonGrid Renders and matches snapshot 1`] = `
<div
  className="wrapper"
  style={
    Object {
      "gridTemplateColumns": "repeat(4, 1fr)",
    }
  }
>
  <button
    className="selected"
    disabled={true}
    key="vFoo"
    onClick={[Function]}
  >
    Foo
  </button>
  <button
    className="selected"
    key="vBar"
    onClick={[Function]}
  >
    Bar
  </button>
  <button
    className=""
    key="vBaz"
    onClick={[Function]}
  >
    Baz
  </button>
  <button
    className=""
    disabled={true}
    key="vQux"
    onClick={[Function]}
  >
    Qux
  </button>
</div>
`;
